### PR TITLE
Add "opening other file formats"

### PR DESCRIPTION
**Pull request by connermcd**
Here's my proposed article. Feel free to edit as desired!

### CODE REVIEW

**Comment by romainl** [comment]
Hi, thank you for this.

I have a few comments:

* having two or three full examples would be cool,
* that PDF example is a bit misleading because `.pdf` is already recognized so there's no need for filetype detection,
* I'd rather have examples involving audio, video, or something fancier,
* you should probably use `~/.vim/after/ftplugin/` instead of `~/.vim/ftplugin/` as it removes the need for the guard,
* adding a gifcast or video would be even cooler.

**Comment by connermcd** [comment]
Can do, I'll get back to you soon.

**Comment by connermcd** [comment]
I've made the suggested changes. The animated gif is somewhat large and may be better hosted elsewhere.

File: content/2018/opening-other-file-formats.md [hunk 1]
```diff
@@ -44,30 +44,38 @@ You can find information for writing your own `ftplugin` files at `:h write-file
...
 and edit it appropriately:
 
-    if exists("b:did_ftplugin")
-      finish
-    endif
-    let b:did_ftplugin = 1
-    sil exe "!zathura " . shellescape(expand("%:p")) . " &" | b# | bd# | redraw! | syn on
+    sil exe "!mplayer " . shellescape(expand("%:p")) . " &>/dev/null &" | b# | bd# | redraw! | syn on
+
+In this example we're using the PDF reader `mplayer` to open the PDF file, but you could just as easily use any PDF viewer. This silently executes an `!external-program` on the current buffer and places it in the background (`&`), discarding any output or errors to `/dev/null`. The `|`s could easily be separated onto multiple lines and act as a list of sequential commands to switch back to the previous buffer (`b#`), delete the PDF buffer (`bd#`), and then ensure that the screen was not garbled in the process by redrawing and ensuring syntax highlighting is on.
```
Comment (romainl) [inline content/2018/opening-other-file-formats.md]: You still have "PDF reader" and "PDF", here.

File: content/2018/opening-other-file-formats.md [hunk 2]
```diff
@@ -44,30 +44,38 @@ You can find information for writing your own `ftplugin` files at `:h write-file
...
+    :!mkdir -p ~/.vim/after/ftplugin/video.vim
 
 and edit it appropriately:
 
-    if exists("b:did_ftplugin")
-      finish
-    endif
-    let b:did_ftplugin = 1
-    sil exe "!zathura " . shellescape(expand("%:p")) . " &" | b# | bd# | redraw! | syn on
+    sil exe "!mplayer " . shellescape(expand("%:p")) . " &>/dev/null &" | b# | bd# | redraw! | syn on
```
Comment (romainl) [inline content/2018/opening-other-file-formats.md]: I think using full command names helps readers. `sil` should be `silent`, `b#` sould be `buffer#`, etc.

Also, wouldn't it be easier to use `:help system()`?

    call system('mplayer path/to/foo.mkv')

And maybe let `xdg-open` and the likes deal with choosing the right handler?

File: content/2018/opening-other-file-formats.md [hunk 3]
```diff
@@ -44,30 +44,38 @@ You can find information for writing your own `ftplugin` files at `:h write-file
...
+
+    :!mkdir -p ~/.vim/ftdetect/audio.vim
+    au BufRead,BufNewFile *.mp3,*.flac,*.wav,*.ogg set filetype=audio
+    :!mkdir -p ~/.vim/after/ftplugin/audio.vim
+    sil exe "!mplayer " . shellescape(expand("%:p")) . " &>/dev/null &" | b# | bd# | redraw! | syn on
+
+Here's an example of opening a video with vim:
 
-In this example we're using the PDF reader `zathura` to open the PDF file, but you could just as easily use any PDF viewer. This silently executes an `!external-program` on the current buffer and places it in the background (`&`). The `|`s could easily be separated onto multiple lines and act as a list of sequential commands to switch back to the previous buffer (`b#`), delete the PDF buffer (`bd#`), and then ensure that the screen was not garbled in the process by redrawing and ensuring syntax highlighting is on.
+![MPlayer](opening-other-file-formats/example.gif)\
```
Comment (romainl) [inline content/2018/opening-other-file-formats.md]: …WE WATCHED THAT VERY SAME EPISODE YESTERDAY…

**Comment by romainl** [comment]
> I've made the suggested changes. The animated gif is somewhat large and may be better hosted elsewhere.

Do you have the original video?

File: content/2018/opening-other-file-formats.md (hunk 1 above)
Comment (connermcd) [reply content/2018/opening-other-file-formats.md]: Woops!

File: content/2018/opening-other-file-formats.md (hunk 2 above)
Comment (connermcd) [reply content/2018/opening-other-file-formats.md]: True, depends on the system you're using. For instance, Mac will use `open`, linux uses `xdg-open` which requires `.desktop` configuration, and windows is a whole different ball game. What would you suggest for cross-platform compatibility?

File: content/2018/opening-other-file-formats.md (hunk 3 above)
Comment (connermcd) [reply content/2018/opening-other-file-formats.md]: Oh you mean that episode of House? hahaha

File: content/2018/opening-other-file-formats.md (hunk 2 above)
Comment (romainl) [reply content/2018/opening-other-file-formats.md]: I use [this snippet](https://github.com/romainl/vim-devdocs/blob/master/plugin/devdocs.vim#L16)

File: content/2018/opening-other-file-formats.md (hunk 3 above)
Comment (connermcd) [reply content/2018/opening-other-file-formats.md]: What would be your ideal video format/size and where should I send/upload it?

**Comment by connermcd** [comment]
Additional changes made as suggested. Please let me know what size/format you'd like the video to be and I'll share/upload it as needed at your discretion. Cheers!

**Comment by romainl** [comment]
> Additional changes made as suggested. Please let me know what size/format you'd like the video to be and I'll share/upload it as needed at your discretion. Cheers!

A 720p-ish MP4 would work. You can send a wetransfer to ${USERNAME} at ${PROJECT_NAME}.

File: content/2018/opening-other-file-formats.md (hunk 3 above)
Comment (romainl) [reply content/2018/opening-other-file-formats.md]: A 720p-ish MP4 would work. You can send a wetransfer to ${USERNAME} at ${PROJECT_NAME}.

**Comment by connermcd** [comment]
Not sure if you're using some kind of command-line github client and bash variables weren't expanded or if you're referring to a tongue-in-cheek username of "${USERNAME}", but here's the wetransfer link: https://we.tl/t-aSD7yi6sxC

**Comment by romainl** [comment]
> Not sure if you're using some kind of command-line github client and bash variables weren't expanded or if you're referring to a tongue-in-cheek username of "${USERNAME}", but here's the wetransfer link: https://we.tl/t-aSD7yi6sxC

An attempt was made. Anyway, thank you for the file.

> Hosting the animated gif also seems like a worthwhile consideration. It's hard on git but light on your server and the viewer's browser.

The GIF weights around 82.7Mb while the MP4 weights 683Kb. Our server, bandwidth, and Git host don't care but our readers in Bengalore (or simply anyone reading us on a mobile) certainly have their preference. The video will be perfect.

**Pull request merged**